src/wallet/wallet.cpp

```diff
@@ -1520,6 +1519,19 @@ struct smallestcoincomp
     }
 };
 
+template <typename I>
+void Shuffle(I first, I last)
+{
+    while (first != last) {
+        uint64_t j = GetRand(last - first);
+        if (j) {
+            using std::swap;
+            swap(*first, *(first + j));
+        }
+        ++first;
+    }
+}
+
```

Comment:
Sigh... sometimes I feel like progress is not progress with the new C++ standards...

Reply:
Also one of the reasons why I don't like C++.

Then again just std::shuffle with the stdlib's random could be used and that might work but I don't exactly know if that'll create any issues.

Reply:
I think the std::shuffle with std::default_random_engine(seed) as the generator will work, right? The seed can use the epoch time.

Reply:
There is nothing particularly sensitive about this shuffle from a wallet point of view as long as it generates a different shuffle order each time it is called.

Reply:
Using `GetAdjustedTime()` as seed fails wallet_tests as the tests are run so fast that the time doesn't change. Using `static_cast<unsigned int>(GetRand(0x80000000))` as seed works but it is a little bit weird imo.

Reply:
GetTimeMillis(), the same medium resolution timer I use for the MilliTimer?

Reply:
Or std::chrono::milliseconds GetTime() if you don't like the underlying use of boost's time classes.

Reply:
There is a microsecond level too if we need it.